napari/_qt/widgets/qt_dims_sorter.py

```diff
@@ -46,55 +36,99 @@ def _array_in_range(arr: np.ndarray, low: int, high: int) -> bool:
     return (arr >= low) & (arr < high)
 
 
-def move_indices(axes_list: SelectableEventedList, order: Tuple[int, ...]):
-    with axes_list.events.blocker_all():
-        if tuple(axes_list) == tuple(order):
+# TODO: This function is in the current usecase of QtDimsSorter unnecessary and could be removed
```

Comment:
If I remember correctly, I think this comment is suggesting that we disconnect this from `dims.events.order` and remove this logic entirely because the `QtDimsSorter` is always recreated each time it is opened, so there is no need to try to keep it consistent with the state in `Dims`.

And trying to keep things consistent is [buggy on this branch, but also on `main` in more limited circumstances](https://github.com/napari/napari/pull/5986#issuecomment-1730494261).

I.e. removing this and the connection logic fixes the bug.

The bigger question is do we want to support the case where we want to keep these in sync? The test I wrote assumes that's the case, but that's not a strong reason. If we do, then we'll need to fix `move_indices`, which may not be too tricky.

Reply:
One potential solution I thought of also deletes this function and instead defines:

```python
    def _dims_order_callback(self, event):
        self.axis_list = AxisList.from_dims(self.dims)
        self.view.setRoot(self.axis_list)
```

where `self.view` stores the `QListView`.

Any thoughts?